consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -458,13 +458,16 @@ private boolean canRetryBlockCreation(final Throwable throwable) {
   @Override
   public Optional<BlockHeader> getOrSyncHeadByHash(final Hash headHash, final Hash finalizedHash) {
     final var chain = protocolContext.getBlockchain();
+    final var chainHead = chain.getChainHeadBlockNumber();
     final var maybeHeadHeader = chain.getBlockHeader(headHash);
 
-    if (maybeHeadHeader.isPresent()) {
-      LOG.atDebug()
+    if (maybeHeadHeader.isPresent()
+        && Math.abs(maybeHeadHeader.get().getNumber() - chainHead) < 500) {
```

Comment:
Can we replace the number here for a named constant to describe where it comes from?

Reply:
Good Catch. Indeed I used this number for the tests but we must definitely use the Max Layer that we have for bonsai

Reply:
https://github.com/hyperledger/besu/blob/ec6a6b1b73751575e3657eff2b837d7083071f37/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/trielog/AbstractTrieLogManager.java#L42

Reply:
isn't that configurable though?

Reply:
Yes there is a flag to change this value

Reply:
if we defer to a WorldStateArchive method that provides this limit, we can skip the behavior for forest since it should not have a worldstate move issue.

Reply:
I removed this part. I will just keep the worldtsate head update modification https://github.com/hyperledger/besu/pull/5699/files#diff-6da2d65dd7a0564527e8ce3ad7c4482f599fb2af30e37316c47854e5aed2e977R610 . I'm doing new test now with this last version